index.js

```diff
@@ -63,25 +64,27 @@ module.exports = function base(app, opts) {
    * var config = {};
    * app.run(config);
    * ```
+   *
    * @name .run
    * @param {Object} `value` Object to be modified by plugins.
    * @return {Object} Returns the object passed to `run`
    * @api public
    */
 
-  utils.define(app, 'run', function(val) {
-    if (!utils.isObject(val)) return;
-    decorate(val);
+   utils.define(app, 'run', function run(val) {
+    if (utils.isObject(val)) {
+      decorate(arguments[0]);
+    }
 
     var self = this || app;
-    var fns = self[prop];
-    var len = fns.length;
-    var idx = -1;
+    var len = self[prop].length;
+    var i = 0;
 
-    while (++idx < len) {
-      val.use(fns[idx]);
+    while (i < len) {
+      var plugin = self[prop][i++];
+      plugin.apply(self, arguments);
```

Comment:
hm. it maybe should still call `.use`, because i think the nesting isn't working now - maybe don't have tests for that?